doc/book/src/user/metrics.md

```diff
@@ -54,14 +54,60 @@ zcash_chain_verified_block_total 162
 ```
 
 By default, access is restricted to localhost. This can be expanded with
-`-metricsallowip=<ip>`, which can specify IPs or subnets. Note that HTTPS is not
-supported, and therefore connections to the endpoint are not encrypted or
+`-metricsallowip=<ip>`, which can specify IPs or subnets. Note that HTTPS is
+not supported, and therefore connections to the endpoint are not encrypted or
 authenticated. Access to the endpoint should be assumed to compromise the
 privacy of node operations, by the provided metrics and/or by timing side
 channels. Non-localhost access is **strongly discouraged** if the node has a
 wallet holding live funds.
 
-### Example metrics collection with Docker
+### Metrics collection with Docker
+
+A docker-compose.yml has been provided in `./contrib/metrics` that sets up
+local instances of `prometheus` and `grafana` and provides a dashboard that
+charts several of the various metrics exposed by `zcashd`'s `prometheus`
+endpoint.
+
+`sudo docker-compose up` will start local instances of `prometheus` and `grafana`,
+accessible on ports `9090` and `3030` respectively. 
+
+```
+cd $(zcash_root)/contrib/metrics
+sudo docker-compose up -d
+```
+
+`~/.zcash/zcash.conf` must be updated to enable `prometheus` and to allow the
+`prometheus` server launched via `docker-compose` to connect to the `zcashd`
+prometheus endpoint. The following script can be used to detect the local IP
+address for the `prometheus` server and add it to the `~/.zcash/zcash.conf` 
+file. 
+
+```
+export PROMETHEUS_DOCKER_IP=$(sudo docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' zcashd-prometheus)
+cat << PROM_CONF >> ~/.zcash/zcash.conf
+prometheusport=9969
+metricsallowip=$PROMETHEUS_DOCKER_IP/32
+PROM_CONF
```

Comment:
Part of the issue with adding this as a script is that its usage varies between Linux and macOS; on macOS docker doesn't require `sudo` to run; also, if it were to be written as a script I think it would need to be able to do more intelligent updates to the ~/.zcash/zcash.conf file. I think I'd rather change the documentation slightly to clarify this issue and just add this as an example of how you might update your config.

Reply:
I've separated out the commands and added some notes, so that people won't be tempted to paste it all as a block.